portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,12 +96,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       return new Comment(id, author, commentText, timestamp);
     }).collect(Collectors.toList());
 
-    // Convert the list of comments to JSON.
-    String json = convertToJson(comments);
+    // Convert the list of comments to a JsonElement.
+    JsonElement commentsJsonElement = convertToJsonElement(comments);
+    json.add("comments", commentsJsonElement);
 
     // Send the JSON as the response.
-    response.setContentType("text/html;");
-    response.getWriter().println(json);
+    response.setContentType("application/json;");
+    response.getWriter().println(json.toString());
+  }
+
+  /** 
+   * Returns the name of the user with id, or null if a name has not been set. 
+   */
+  private String getUserName(String id) {
```

Comment:
You could consider returning an Optional<String> as it removes ambiguity about a reference's nullability (although this is fine for a private method).